.github/workflows/release.yml

```diff
@@ -25,7 +29,7 @@ jobs:
 
       - name: Login to GitHub Container Registry
         uses: docker/login-action@v2
-        if: startsWith(inputs.tag, 'v')
+        if: startsWith(inputs.tag, 'v') || (inputs.tag == 'main' && inputs.prerelease)
```

Comment:
It may be worth pulling out `(inputs.tag == 'main' && inputs.prerelease)` to it's own variable. This way you can use it throughout the workflow easier and it improves readability.

right now I'm noticing there's areas where we miss the && `inputs.tag == 'main'